// optimizely-oui/dist/js/components/Token/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Button = require('../Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DismissButton = function DismissButton(onDismiss, testSection) {
  return _react2.default.createElement(
    _Button2.default,
    {
      onClick: onDismiss,
      style: 'plain',
      size: 'tiny',
      ariaLabel: 'Close alert',
      testSection: testSection ? testSection + '-dismiss' : null },
    '×'
  );
};

/**
 * Token to be used to make token lists.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */
var Token = function Token(props) {
  return _react2.default.createElement(
    'div',
    {
      className: 'oui-token-wrap',
      'data-test-section': props.testSection },
    _react2.default.createElement(
      'div',
      { className: 'oui-token oui-token--' + props.style },
      props.name
    ),
    props.isDismissible ? DismissButton(props.onDismiss, props.testSection) : null
  );
};

Token.propTypes = {
  /** Determines if token has dismissible feature or not */
  isDismissible: _react2.default.PropTypes.bool,
  /** Name label on token */
  name: _react2.default.PropTypes.string.isRequired,
  /** Function to call when token has been dismissed */
  onDismiss: _react2.default.PropTypes.func,
  /** Determines style of token depending on priority level */
  style: _react2.default.PropTypes.oneOf(['primary', 'secondary']),
  /** Hook for automated JavaScript tests */
  testSection: _react2.default.PropTypes.string
};

Token.defaultProps = {
  isDismissible: false,
  style: 'secondary'
};

exports.default = Token;
