// optimizely-oui/dist/js/components/Table/example/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('../index');

var _index2 = _interopRequireDefault(_index);

var _Button = require('../../Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = [{
  isPadded: true,
  examples: [_react2.default.createElement(_index2.default, {
    data: [['Header CTA', '12%', 'Paused'], ['Shorter Contact Form', '4%', 'Draft'], ['Larger search bar', '6.7%', 'Paused'], ['Center aligned headline', '9.3%', 'Running']],
    headings: ['Experiment', 'Conversion Rate', 'Status'] })]
}, {
  isPadded: true,
  examples: [_react2.default.createElement(_index2.default, {
    data: [['Header CTA', '12%', 'Running'], ['Shorter Contact Form', '4%', 'Draft'], ['Larger search bar', '6.7%', 'Paused'], ['Center aligned headline', '9.3%', 'Running']],
    headings: ['Experiment', 'Conversion Rate', 'Status'],
    numberedColumns: [1],
    style: 'rule' })]
}, {
  isPadded: true,
  examples: [_react2.default.createElement(_index2.default, {
    data: [['Header CTA', '12%', _react2.default.createElement(
      _Button2.default,
      { size: 'small' },
      'Start'
    )], ['Shorter Contact Form', '4%', _react2.default.createElement(
      _Button2.default,
      { size: 'small' },
      'Start'
    )], ['Larger search bar', '6.7%', _react2.default.createElement(
      _Button2.default,
      { size: 'small' },
      'Start'
    )], ['Center aligned headline', '9.3%', _react2.default.createElement(
      _Button2.default,
      { size: 'small' },
      'Start'
    )]],
    headings: ['Experiment', 'Conversion Rate', 'Status'],
    numberedColumns: [1],
    collapsedColumns: [2],
    style: 'wall' })]
}, {
  isPadded: true,
  examples: [_react2.default.createElement(_index2.default, {
    data: [['Header CTA', '12%', 'Running'], ['Shorter Contact Form', '4%', 'Draft'], ['Larger search bar', '6.7%', 'Paused'], ['Center aligned headline', '9.3%', 'Running']],
    style: 'wall' })]
}]; /* eslint-disable react/jsx-key */
