// optimizely-oui/dist/js/components/Attention/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Button = require('../Button');

var _Button2 = _interopRequireDefault(_Button);

var _accessibility = require('../../utils/accessibility');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var renderDismissButton = function renderDismissButton(testSection) {
  return _react2.default.createElement(
    'div',
    { className: 'oui-attention__close' },
    _react2.default.createElement(
      _Button2.default,
      {
        style: 'plain',
        size: 'small',
        ariaLabel: 'Close alert',
        testSection: testSection + '-dismiss' },
      '×'
    )
  );
};

/**
 * Provide contextual feedback with a wide range of styles that can be used to
 * convey the message's importance.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */
var Attention = function Attention(props) {
  var colorClassName = props.type ? 'oui-attention--' + props.type : '';
  var alignmentClassName = props.alignment === 'center' ? 'oui-text--center' : '';
  var attentionAriaLabel = props.type ? (0, _accessibility.getAssistiveTextFromColorClass)(props.type) : null;
  var classes = ('oui-attention ' + colorClassName + ' ' + alignmentClassName).trim();

  return _react2.default.createElement(
    'div',
    {
      className: classes,
      'data-test-section': props.testSection,
      'aria-label': attentionAriaLabel,
      role: 'alert' },
    props.isDismissible ? renderDismissButton(props.testSection) : null,
    props.children
  );
};

Attention.propTypes = {
  /** How the text is aligned */
  alignment: _react2.default.PropTypes.oneOf(['left', 'center']),
  /** Text that appears within the component */
  children: _react2.default.PropTypes.string.isRequired,
  /** Button allowing users to dismiss the component */
  isDismissible: _react2.default.PropTypes.bool,
  /** Hook for automated JavaScript tests */
  testSection: _react2.default.PropTypes.string,
  /** Various color schemes */
  type: _react2.default.PropTypes.oneOf(['bad-news', 'brand', 'good-news', 'warning'])
};

exports.default = Attention;
