// optimizely-oui/dist/js/components/ButtonRow/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ButtonRowSide = function ButtonRowSide(side, buttons, testSection) {
  return _react2.default.createElement(
    'div',
    {
      className: 'flex--1 oui-button-row--' + side,
      'data-test-section': testSection ? testSection + '-' + side : null },
    buttons
  );
};

/**
 * A horizontal list of buttons with space between them.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */
var ButtonRow = function ButtonRow(props) {
  return _react2.default.createElement(
    'div',
    {
      className: 'oui-button-row flex',
      'data-test-section': props.testSection },
    props.leftGroup ? ButtonRowSide('left', props.leftGroup, props.testSection) : null,
    props.rightGroup ? ButtonRowSide('right', props.rightGroup, props.testSection) : null
  );
};

ButtonRow.propTypes = {
  /** Array of buttons for left side */
  leftGroup: _react2.default.PropTypes.arrayOf(_react2.default.PropTypes.element),
  /** Array of buttons for right side */
  rightGroup: _react2.default.PropTypes.arrayOf(_react2.default.PropTypes.element),
  /** Hook for automated JavaScript tests */
  testSection: _react2.default.PropTypes.string
};

exports.default = ButtonRow;
