// optimizely-oui/dist/js/utils/test-helpers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.simulate = exports.expectTestSectionToNotExist = exports.expectTestSectionToExist = exports.getTestSectionFromComponent = exports.getNodeFromComponent = exports.renderIntoDocument = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _reactAddonsTestUtils = require('react-addons-test-utils');

var _reactAddonsTestUtils2 = _interopRequireDefault(_reactAddonsTestUtils);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Wraps a component in a `div` and calls a function, provided by React, to
 * render it into the document. This is needed for stateless components.
 * https://github.com/facebook/react/issues/4839
 * @param {ReactElement} component - JSX returned in a function
 * @returns {ReactElement} Orignal component wrapped in a `div` and rendered
 * into the document.
 */
var renderIntoDocument = exports.renderIntoDocument = function renderIntoDocument(component) {
  return _reactAddonsTestUtils2.default.renderIntoDocument(_react2.default.createElement(
    'div',
    null,
    component
  ));
};

/**
 * Grab the first child of a ReactElement. This is useful after calling the
 * `renderIntoDocument` helper function that wraps a component in a `div`. This
 * is needed because of a limitation of stateless React components.
 * https://github.com/facebook/react/issues/4839
 * @param {ReactElement} component - JSX returned in a function
 * @returns {HTMLElement} Orignal component wrapped in a `div` and rendered
 * into the document.
 */
var getNodeFromComponent = exports.getNodeFromComponent = function getNodeFromComponent(component) {
  return _reactDom2.default.findDOMNode(component).children[0];
};

/**
 * Find an element that contains an attribute `data-test-section equal` to the
 * provided value.
 * @param {ReactElement} component - JSX returned in a function
 * @param {String} testSection - unique label identifying an element
 * @returns {HTMLElement} Element that has the provided data test section.
 */
var getTestSectionFromComponent = exports.getTestSectionFromComponent = function getTestSectionFromComponent(component, testSection) {
  var componentNode = getNodeFromComponent(component);
  var element = componentNode.querySelector('[data-test-section="' + testSection + '"]');

  // Check test-section on parent if missing from children
  if (!element && componentNode.getAttribute('data-test-section') === testSection) {
    element = componentNode;
  }

  return element;
};

/**
 * Assert that an HTMLElement has an attribute `data-test-section` of value
 * `testSection`.
 * @param {HTMLElement} componentNode - Element that should contain test section
 * @param {String} testSection - value of the `data-test-section`
 * @returns {undefined}
 */
var expectTestSectionToExist = exports.expectTestSectionToExist = function expectTestSectionToExist(componentNode, testSection) {
  var val = componentNode.getAttribute('data-test-section');

  if (val !== testSection) {
    throw new Error(val + ' does not equal ' + testSection);
  }
};

/**
 * Assert that an HTMLElement does not have an attribute `data-test-section`
 * of value `testSection`.
 * @param {HTMLElement} componentNode - Element that should not contain test
 * section
 * @param {String} testSection - value of the `data-test-section`
 * @returns {undefined}
 */
var expectTestSectionToNotExist = exports.expectTestSectionToNotExist = function expectTestSectionToNotExist(componentNode, testSection) {
  var val = componentNode.getAttribute('data-test-section');

  if (val === testSection) {
    throw new Error(val + ' should not equal ' + testSection);
  }
};

/**
 * Simulate events on react components.
 * @returns {Object} Object with methods to simulate events (eg. change, input,
 * etc).
 */
var simulate = exports.simulate = _reactAddonsTestUtils2.default.Simulate;
